spec/stack_spec.rb

```diff
@@ -117,6 +118,69 @@
     expect(@logger).to have_received(:info).with(/Deleting #{name} stack.../)
   end
 
+  context "#parameters_for_update" do
+    it "uses defaults, volatile, use_previous_value" do
+      stack = new_stack(name: "spec-aws-ruby-stack-param-update",
+                        filename: "updating_parameters/orig.yml",
+                        overrides: {
+                          parameter_defaults: {
+                            will_go_away: "value1",
+                            sticks_around: "value2",
+                            volatile_value: "gonna_change"
+                          },
+                          volatile_parameters_block: Proc.new do
+                            volatile_value { name } # the stack's name as a placeholder for stack-specific data
+                          end
+                        })
+      stack.create(params: { sticks_around_no_default: "value3" }, wait: true)
+
+      stack = new_stack(name: "spec-aws-ruby-stack-param-update",
+                        filename: "updating_parameters/mod.yml",
+                        overrides: {
+                          parameter_defaults: {
+                            will_go_away: "value1",
+                            sticks_around: "value2",
+                            added: "added value",
+                            volatile_value: "gonna_change"
+                          },
+                          volatile_parameters_block: Proc.new do
+                            volatile_value { name } # the stack's name
+                          end
+                        })
+
+      # Here is where volatile_value could be changed manually on the tag value
+      # so we don't want to just "use_previous_value" b/c that would be "gonna_change"
+
+      parameters = stack.parameters_for_update(overrides: {sticks_around: "override"})
+
+      expect(parameters).to eq({
+        sticks_around: "override",
+        added: "added value",
+        sticks_around_no_default: :use_previous_value,
+        volatile_value: "spec-aws-ruby-stack-param-update"
+      })
+
+      stack.delete
+    end
+  end
+
+  it "can be created without an absolute_template_path" do
+    expect{OpenStax::Aws::Stack.new(name: "foo", region: "us-east-2")}.not_to raise_error
+  end
+
+  it "can get a existing stack's template without specifying a file" do
+    name = "spec-aws-ruby-stack-existing-template"
+
+    stack = new_template_one_stack(name: name)
+    file_template_contents = stack.template.body
+    stack.create(params: {bucket_name: bucket_name, tag_value: "howdy"}, wait: true)
+
+    stack = described_class.new(name: name, region: region, dry_run: false)
+    expect(stack.template.body).to eq file_template_contents
+
+    stack.delete
```

Comment:
if this tests raises an exception...ok to not delete the stack?

Reply:
This definitely isn't as tidying-up-after-itself as it could be. if the tests pass the stack is deleted.  if the tests fail, the stack isn't deleted but then the spec won't run again until it is deleted b/c AWS will say "stack already exists!" ... so to get the tests to pass you have to manually delete orphaned stacks, and once they pass it will have cleaned up.